### ISSUE TITLE
Garbage Collection Fights with New Download

### ISSUE THREAD

**Issue by uncleJim21** [stakwork/sphinx-ios#229]
Currently garbage collection is set to remove the oldest items in the library if the user exceeds his/her memory limit.

The issue is that suppose they want to download a podcast from 10 years ago but they are almost at the memory limit. What would happen is that as soon as they download the podcast, it would delete it immediately.

Because the current rule we have is assign the "date" value of the storage item as the publish date of the podcast we will have a subpar UX in this scenario. I think we need to update our policy for assigning dates in view of this corner case. @tomastiminskas

**Comment by tomastiminskas** [comment]
@uncleJim21 good catch! I think using the item date works well for chat media items, but maybe for podcasts we can add an attribute in the feed item model (something like `downloaded_at: Date`) and consider that date in the garbage collection logic. Or we can add that attribute in the CachedMedia model and use the item date for chat media items and the current download date for feed items. Thoughts?

**Comment by uncleJim21** [comment]
Yea I think this might be the only way to solve the problem lol
